core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -136,96 +200,235 @@ function getApiLogsData(payload) {
+                  data.timestamp = new Date(1 * logEvent.timestamp).toISOString();
+                  let message = logEvent.message;
+                  let messageLength = Buffer.byteLength(message, 'utf8');
+                  if (messageLength > 32766) { //since 32766(32KB) is the default message size
+                    let truncatedMessage = truncate(message, 32740); // message size + ...[TRUNCATED]
+                    data.message = truncatedMessage + "  ...[TRUNCATED]";
+                  } else {
+                    data.message = message;
+                  }
+
+                  data.log_level = getSubInfo(logEvent.message, config.PATTERNS.log_level, 0);
+                  if (!data.log_level) {
+                    data.log_level = config.DEFAULT_LOG_LEVEL;
+                  }
+
+                  if (!(data.message.startsWith("REPORT") || data.message.startsWith("START") || data.message.startsWith("END"))) {
+                    let timestmp = getSubInfo(data.message, config.PATTERNS.timestamp_pattern, 0);
+                    data.message = data.message.replace(timestmp, "");
+
+                    let guid = getSubInfo(data.message, config.PATTERNS.guid_regex, 0);
+                    data.message = data.message.replace(guid, "");
+
+                    data.message = data.message.replace(data.log_level, "");
+                  }
+                  data.message = data.message.trim();
+                  let indexName = "applicationlogs";
+                  let action = {
+                    "index": {}
+                  };
+                  action.index._index = indexName;
+                  action.index._type = "applicationlogs";
+                  action.index._id = logEvent.id;
+
+                  bulkRequestBody += [
+                    JSON.stringify(action),
+                    JSON.stringify(data),
+                  ].join('\n') + '\n';
+                });
+              } else {
+                logger.error("invalid lambda logs event..: " + JSON.stringify(payload));
+              }
+              callback(bulkRequestBody);
+            } else {
+              callback(null);
+            }
+          })
+        })
+      })
+    })
+  })
+}
 
-        data.platform_log_group = payload.logGroup;
-        data.platform_log_stream = payload.logStream;
-        data.timestamp = new Date(1 * logEvent.timestamp).toISOString();
-        let message = logEvent.message;
-        let messageLength = Buffer.byteLength(message, 'utf8');
-        if (messageLength > 32766) { //since 32766(32KB) is the default message size
-          let truncatedMessage = truncate(message, 32740); // message size + ...[TRUNCATED]
-          data.message = truncatedMessage + "  ...[TRUNCATED]";
+// Function to get service metadata using service API
+function getsServiceMetaData(config, logGroup, authToken) {
+  var serviceParts = logGroup.split('_');
+  if(serviceParts.length < 2){
```

Comment:
Which use case needs this?

Reply:
This is the case for api/lambda build packs. Since the log Group for this respective build packs are coming with '-' and not '_', so to handle those scenarios.

Reply:
Hmm, it's better to change the log group naming convention in build packs instead of this hacky fix :)

Reply:
Also, Get rid of `updatePolicy` function in the build packs, destination policy updates are being done during install time already and overriding it with each build can break how logs are being ingested - https://github.com/tmobile/jazz/blob/node10/builds/jenkins-build-pack-function/Jenkinsfile#L874.

@raghits FYI.

Reply:
refactored, the above code for hack is removed and changes done to api and lambda build pack and supporting changes done to serverless-*.yml